staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -405,63 +393,16 @@ func (cnc *CloudNodeController) initializeNode(ctx context.Context, node *v1.Nod
 // a node object with provider-specific information.
 // All of the returned functions are idempotent, because they are used in a retry-if-conflict
 // loop, meaning they could get called multiple times.
-func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(ctx context.Context, node *v1.Node) ([]nodeModifier, error) {
-	var (
-		nodeModifiers []nodeModifier
-		providerID    string
-		err           error
-	)
-
+func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(
+	ctx context.Context,
+	node *v1.Node,
+	providerID string,
+	instanceMeta *cloudprovider.InstanceMetadata,
+) ([]nodeModifier, error) {
+
+	var nodeModifiers []nodeModifier
 	if node.Spec.ProviderID == "" {
```

Comment:
There also needs to be a check for when the passed in providerID is "" which can happen if the provider returns `cloudprovider.NotImplemented` like in line 423

Reply:
`if node.Spec.ProviderID == "" && providerID != ""`, right?

Reply:
Done.